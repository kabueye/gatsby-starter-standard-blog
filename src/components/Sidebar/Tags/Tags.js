// @flow strict
import React from 'react';
import kebabCase from 'lodash/kebabCase';
import { Link } from 'gatsby';
import styles from './Tags.module.scss';
import { useTagsList } from '../../../hooks';

const Tags = () => {
  const tags = useTagsList();

  return (<div className={styles['tags']}>
      <div className={styles['tags__inner']}>
        <div className={styles['tags__title']}>Tags</div>
        <ul>
          {tags.map((tag) => (
            <li key={tag.fieldValue}>
              <Link to={`/tag/${kebabCase(tag.fieldValue)}/`}>
                {tag.fieldValue} ({tag.totalCount})
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default Tags;
