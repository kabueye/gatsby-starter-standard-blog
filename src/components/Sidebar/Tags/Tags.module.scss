@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.tags {
    @include margin-bottom(1);
  
    &__title {
      font-size: $typographic-base-font-size * 1.2;
      font-weight: bold;
     }
  }