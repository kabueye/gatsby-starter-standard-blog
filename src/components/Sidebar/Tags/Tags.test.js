// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Tags from './Tags';
import allMarkdownRemark from '../../../../jest/__fixtures__/all-markdown-remark';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';

describe('Tags', () => {
  const props = {
    ...siteMetadata,
    ...allMarkdownRemark
  };

  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(props)
      ),
      useStaticQuery.mockReturnValue(props)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Tags />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
