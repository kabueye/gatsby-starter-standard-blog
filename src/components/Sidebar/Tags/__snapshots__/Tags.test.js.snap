// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div
  className="tags"
>
  <div
    className="tags__inner"
  >
    <div
      className="tags__title"
    >
      Tags
    </div>
    <ul>
      <li>
        <a
          href="/tag/test-0/"
        >
          test_0
           (
          1
          )
        </a>
      </li>
      <li>
        <a
          href="/tag/test-1/"
        >
          test_1
           (
          2
          )
        </a>
      </li>
    </ul>
  </div>
</div>
`;
